Index: .idea/jarRepositories.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/jarRepositories.xml b/.idea/jarRepositories.xml
new file mode 100644
--- /dev/null	(date 1727652138636)
+++ b/.idea/jarRepositories.xml	(date 1727652138636)
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Central Repository" />
+      <option name="url" value="https://repo.maven.apache.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="sonatype" />
+      <option name="name" value="sonatype" />
+      <option name="url" value="https://oss.sonatype.org/content/groups/public/" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="spigotmc-repo" />
+      <option name="name" value="spigotmc-repo" />
+      <option name="url" value="https://hub.spigotmc.org/nexus/content/repositories/snapshots/" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Maven Central repository" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
+    </remote-repository>
+    <remote-repository>
+      <option name="id" value="jboss.community" />
+      <option name="name" value="JBoss Community repository" />
+      <option name="url" value="https://repository.jboss.org/nexus/content/repositories/public/" />
+    </remote-repository>
+  </component>
+</project>
\ No newline at end of file
Index: target/maven-archiver/pom.properties
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/target/maven-archiver/pom.properties b/target/maven-archiver/pom.properties
new file mode 100644
--- /dev/null	(date 1727663908124)
+++ b/target/maven-archiver/pom.properties	(date 1727663908124)
@@ -0,0 +1,3 @@
+artifactId=EndsWhenOnePersonDies
+groupId=jp.klneg.dev
+version=1.0-Remo
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1727665878986)
+++ b/.idea/vcs.xml	(date 1727665878986)
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/.gitignore b/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1727652139119)
+++ b/.idea/.gitignore	(date 1727652139119)
@@ -0,0 +1,8 @@
+# Default ignored files
+/shelf/
+/workspace.xml
+# Editor-based HTTP Client requests
+/httpRequests/
+# Datasource local storage ignored files
+/dataSources/
+/dataSources.local.xml
Index: target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst b/target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst
new file mode 100644
--- /dev/null	(date 1727663907593)
+++ b/target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst	(date 1727663907593)
@@ -0,0 +1,4 @@
+jp\klneg\dev\endsWhenOnePersonDies\Messanger\PlayerTransfer.class
+jp\klneg\dev\endsWhenOnePersonDies\Scheduler\ServerRestartScheduler.class
+jp\klneg\dev\endsWhenOnePersonDies\EndsWhenOnePersonDies.class
+jp\klneg\dev\endsWhenOnePersonDies\EventHandler\PlayerDeathEventListener.class
Index: src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EventHandler/PlayerDeathEventListener.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EventHandler/PlayerDeathEventListener.java b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EventHandler/PlayerDeathEventListener.java
new file mode 100644
--- /dev/null	(date 1727663284225)
+++ b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EventHandler/PlayerDeathEventListener.java	(date 1727663284225)
@@ -0,0 +1,33 @@
+package jp.klneg.dev.endsWhenOnePersonDies.EventHandler;
+
+import jp.klneg.dev.endsWhenOnePersonDies.Scheduler.ServerRestartScheduler;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.GameMode;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.plugin.Plugin;
+
+import static jp.klneg.dev.endsWhenOnePersonDies.EndsWhenOnePersonDies.deathCount;
+
+public class PlayerDeathEventListener implements Listener {
+    Plugin plugin;
+
+    public PlayerDeathEventListener(Plugin plugin) {
+        this.plugin = plugin;
+    }
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent event) {
+        Player player = event.getEntity();
+        deathCount++;
+        for (Player p : plugin.getServer().getOnlinePlayers()) {
+            p.sendTitle("死んでしまった......", deathCount + "回目の死亡です。", 10, 60, 10);
+            p.setGameMode(GameMode.SPECTATOR);
+        }
+        Bukkit.broadcastMessage(ChatColor.GOLD + "30秒後に再起動をします.....");
+        new ServerRestartScheduler().runTaskTimer(plugin, 0L, 20L);
+    }
+}
Index: target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst b/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst
new file mode 100644
--- /dev/null	(date 1727663906930)
+++ b/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst	(date 1727663906930)
@@ -0,0 +1,4 @@
+C:\Users\cyrus\IdeaProjects\MCBAD\EndsWhenOnePersonDies\src\main\java\jp\klneg\dev\endsWhenOnePersonDies\EndsWhenOnePersonDies.java
+C:\Users\cyrus\IdeaProjects\MCBAD\EndsWhenOnePersonDies\src\main\java\jp\klneg\dev\endsWhenOnePersonDies\EventHandler\PlayerDeathEventListener.java
+C:\Users\cyrus\IdeaProjects\MCBAD\EndsWhenOnePersonDies\src\main\java\jp\klneg\dev\endsWhenOnePersonDies\Messanger\PlayerTransfer.java
+C:\Users\cyrus\IdeaProjects\MCBAD\EndsWhenOnePersonDies\src\main\java\jp\klneg\dev\endsWhenOnePersonDies\Scheduler\ServerRestartScheduler.java
Index: src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Scheduler/ServerRestartScheduler.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Scheduler/ServerRestartScheduler.java b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Scheduler/ServerRestartScheduler.java
new file mode 100644
--- /dev/null	(date 1727663700530)
+++ b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Scheduler/ServerRestartScheduler.java	(date 1727663700530)
@@ -0,0 +1,39 @@
+package jp.klneg.dev.endsWhenOnePersonDies.Scheduler;
+
+import jp.klneg.dev.endsWhenOnePersonDies.Messanger.PlayerTransfer;
+import net.md_5.bungee.api.ChatMessageType;
+import net.md_5.bungee.api.chat.TextComponent;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitRunnable;
+
+import static jp.klneg.dev.endsWhenOnePersonDies.EndsWhenOnePersonDies.plugin;
+
+public class ServerRestartScheduler extends BukkitRunnable {
+
+
+    private int nowTime;
+
+    public ServerRestartScheduler() {
+        this.nowTime = 30;
+    }
+
+    @Override
+    public void run() {
+        if (nowTime < 0) {
+            for (Player p : Bukkit.getOnlinePlayers()) {
+                PlayerTransfer.start(p, plugin);
+            }
+            plugin.getServer().shutdown();
+            return;
+        }
+        if (nowTime % 10 == 0) {
+            plugin.getLogger().info("CountDown:" + nowTime);
+        }
+        for (Player p : Bukkit.getOnlinePlayers()) {
+            p.spigot().sendMessage(ChatMessageType.ACTION_BAR, new TextComponent("あと" + nowTime + "秒"));
+        }
+        nowTime--;
+    }
+
+}
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
new file mode 100644
--- /dev/null	(date 1727652136554)
+++ b/pom.xml	(date 1727652136554)
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>jp.klneg.dev</groupId>
+    <artifactId>EndsWhenOnePersonDies</artifactId>
+    <version>1.0-Remo</version>
+    <packaging>jar</packaging>
+
+    <name>EndsWhenOnePersonDies</name>
+
+    <properties>
+        <java.version>21</java.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <build>
+        <defaultGoal>clean package</defaultGoal>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.13.0</version>
+                <configuration>
+                    <source>${java.version}</source>
+                    <target>${java.version}</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>3.5.3</version>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+    </build>
+
+    <repositories>
+        <repository>
+            <id>spigotmc-repo</id>
+            <url>https://hub.spigotmc.org/nexus/content/repositories/snapshots/</url>
+        </repository>
+        <repository>
+            <id>sonatype</id>
+            <url>https://oss.sonatype.org/content/groups/public/</url>
+        </repository>
+    </repositories>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.spigotmc</groupId>
+            <artifactId>spigot-api</artifactId>
+            <version>1.21.1-R0.1-SNAPSHOT</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+</project>
Index: src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Messanger/PlayerTransfer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Messanger/PlayerTransfer.java b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Messanger/PlayerTransfer.java
new file mode 100644
--- /dev/null	(date 1727655199540)
+++ b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/Messanger/PlayerTransfer.java	(date 1727655199540)
@@ -0,0 +1,27 @@
+package jp.klneg.dev.endsWhenOnePersonDies.Messanger;
+
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+
+import static jp.klneg.dev.endsWhenOnePersonDies.EndsWhenOnePersonDies.lobbyName;
+
+public class PlayerTransfer {
+
+    public static void start(Player player, Plugin plugin) {
+         byte[] data = init();
+         player.sendPluginMessage(plugin, "BungeeCord", data);
+         return;
+    }
+
+
+
+    private static byte[] init() {
+        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+        out.writeUTF("Connect");
+        out.writeUTF(lobbyName);
+
+        return out.toByteArray();
+    }
+}
Index: src/main/resources/config.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/config.yml b/src/main/resources/config.yml
new file mode 100644
--- /dev/null	(date 1727655199536)
+++ b/src/main/resources/config.yml	(date 1727655199536)
@@ -0,0 +1,2 @@
+lobbyName: lobby
+PlayerDeathCounter: 0
\ No newline at end of file
Index: src/main/resources/plugin.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
new file mode 100644
--- /dev/null	(date 1727652136559)
+++ b/src/main/resources/plugin.yml	(date 1727652136559)
@@ -0,0 +1,6 @@
+name: EndsWhenOnePersonDies
+version: '1.0-Remo'
+main: jp.klneg.dev.endsWhenOnePersonDies.EndsWhenOnePersonDies
+api-version: '1.21'
+authors: [ Cyrus ]
+description: HardCore automatic progression
Index: src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EndsWhenOnePersonDies.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EndsWhenOnePersonDies.java b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EndsWhenOnePersonDies.java
new file mode 100644
--- /dev/null	(date 1727657062460)
+++ b/src/main/java/jp/klneg/dev/endsWhenOnePersonDies/EndsWhenOnePersonDies.java	(date 1727657062460)
@@ -0,0 +1,33 @@
+package jp.klneg.dev.endsWhenOnePersonDies;
+
+import jp.klneg.dev.endsWhenOnePersonDies.EventHandler.PlayerDeathEventListener;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public final class EndsWhenOnePersonDies extends JavaPlugin {
+    public static Plugin plugin;
+    public static String lobbyName;
+    public static int deathCount;
+
+    @Override
+    public void onEnable() {
+        // Plugin startup logic
+        saveDefaultConfig();
+        deathCount = getConfig().getInt("PlayerDeathCounter");
+
+
+        getLogger().info("EWOPD Plugin enabled");
+        getServer().getMessenger().registerOutgoingPluginChannel(this, "BungeeCord");
+        getServer().getPluginManager().registerEvents(new PlayerDeathEventListener(this), this);
+        lobbyName = getConfig().getString("lobbyName");
+        plugin = this;
+
+    }
+
+    @Override
+    public void onDisable() {
+        // Plugin shutdown logic
+        getConfig().set("PlayerDeathCounter", deathCount);
+        saveConfig();
+    }
+}
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1727652134584)
+++ b/.idea/misc.xml	(date 1727652134584)
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ExternalStorageConfigurationManager" enabled="true" />
+  <component name="MavenProjectsManager">
+    <option name="originalFiles">
+      <list>
+        <option value="$PROJECT_DIR$/pom.xml" />
+      </list>
+    </option>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="corretto-22" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/modules.xml b/.idea/modules.xml
new file mode 100644
--- /dev/null	(date 1727652138594)
+++ b/.idea/modules.xml	(date 1727652138594)
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/EndsWhenOnePersonDies.iml" filepath="$PROJECT_DIR$/EndsWhenOnePersonDies.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: target/classes/config.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/target/classes/config.yml b/target/classes/config.yml
new file mode 100644
--- /dev/null	(date 1727663906786)
+++ b/target/classes/config.yml	(date 1727663906786)
@@ -0,0 +1,2 @@
+lobbyName: lobby
+PlayerDeathCounter: 0
\ No newline at end of file
Index: EndsWhenOnePersonDies.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/EndsWhenOnePersonDies.iml b/EndsWhenOnePersonDies.iml
new file mode 100644
--- /dev/null	(date 1727652138578)
+++ b/EndsWhenOnePersonDies.iml	(date 1727652138578)
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
+  <component name="FacetManager">
+    <facet type="minecraft" name="Minecraft">
+      <configuration>
+        <autoDetectTypes>
+          <platformType>SPIGOT</platformType>
+        </autoDetectTypes>
+        <projectReimportVersion>1</projectReimportVersion>
+      </configuration>
+    </facet>
+  </component>
+</module>
\ No newline at end of file
Index: .idea/encodings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/encodings.xml b/.idea/encodings.xml
new file mode 100644
--- /dev/null	(date 1727652138627)
+++ b/.idea/encodings.xml	(date 1727652138627)
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Encoding">
+    <file url="file://$PROJECT_DIR$/src/main/java" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/src/main/resources" charset="UTF-8" />
+  </component>
+</project>
\ No newline at end of file
Index: .idea/compiler.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/compiler.xml b/.idea/compiler.xml
new file mode 100644
--- /dev/null	(date 1727652138619)
+++ b/.idea/compiler.xml	(date 1727652138619)
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="CompilerConfiguration">
+    <annotationProcessing>
+      <profile name="Maven default annotation processors profile" enabled="true">
+        <sourceOutputDir name="target/generated-sources/annotations" />
+        <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
+        <outputRelativeToContentRoot value="true" />
+        <module name="EndsWhenOnePersonDies" />
+      </profile>
+    </annotationProcessing>
+  </component>
+</project>
\ No newline at end of file
Index: target/classes/plugin.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/target/classes/plugin.yml b/target/classes/plugin.yml
new file mode 100644
--- /dev/null	(date 1727663906789)
+++ b/target/classes/plugin.yml	(date 1727663906789)
@@ -0,0 +1,6 @@
+name: EndsWhenOnePersonDies
+version: '1.0-Remo'
+main: jp.klneg.dev.endsWhenOnePersonDies.EndsWhenOnePersonDies
+api-version: '1.21'
+authors: [ Cyrus ]
+description: HardCore automatic progression
